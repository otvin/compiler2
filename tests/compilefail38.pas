program compilefail38(output);
{listing same constant multiple times as value of an enumerated type}

type fruit = (apple, pear, banana, cherry, apple);

var myfruit:fruit;

begin
	myfruit := cherry;
	writeln('it''s cherry');
	myfruit := pred(myfruit);
	writeln('now it''s banana');
	myfruit := pred(myfruit);
	writeln('now it''s pear');
	myfruit := pred(myfruit);
	writeln('now it''s apple');
	writeln('next line should fail with an error');
	myfruit := pred(myfruit);
	writeln('we shouldn''t get here.');

end.
