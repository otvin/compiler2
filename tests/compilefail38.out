Identifier redefined: 'apple' 'apple' in file:tests/compilefail38.pas line:4     column:44 
